### PR TITLE
Refactoring Mobile user menu

### PR DESCRIPTION
**Pull request by Ruud68**
Pull Request for Issue #9306 #9299 replacement for #9300  . 
 
#### Summary of Changes 
The way the Kunena menu is implemented is not conform BS5 best practises: the menu has two parts a collapse part with the menu items in it, and the part where the login / logout button is displayed.

It turns out that when the menu is collapsed, not only the collapsed menu is in the collapse, but also the login / logout buttons, these are also still shown on the main (non-collapsed) menu bar.

I have been trying to fix that but this was not possible. I have now refactored the menu to a offcanvas menu.
it turns out that the offcanvas part is more strict: it doesn't display the login / logout buttons: this is exactly what we want.
An IMO it also looks better :)

this PR also contans the fix that login in on the mobile menu (dropdown) wasn't possible (the username field was set as email instead of text input).
 
#### Testing Instructions
on device type mobile, tablet, desktop test if login / login toggles the dropdown and if login / logout via dropdown works.
on mobile and tablet test if offcanvas menu shows / hides, test if all menu items also work.
test if dropdown for login / logout are always in the viewport (and not partially offscreen)

cc @rich20

### CODE REVIEW

**Comment by xillibit** [comment]
I had rebased the branch K6.0 to rename some commits, becaus the commit message should start whith [K6.0] if it's on K6.0 branch.

Is-it possible that you restart by pulling K6.0 and delete the existing K6.0 ?

**Comment by Ruud68** [comment]
@xillibit done, I hope... #9310

File: src/site/template/aurelia/layouts/widget/menubar/default.php [hunk 1]
```diff
@@ -12,26 +12,29 @@
 
 defined('_JEXEC') or die();
 
-use Kunena\Forum\Libraries\Icons\KunenaIcons;
+use Joomla\CMS\Language\Text;
 
 ?>
```
Comment (fotofreunde) [inline src/site/template/aurelia/layouts/widget/menubar/default.php:18]: Hi!
I do not know what I'm doing wrong ;)
If I change that, I'll lose my registration. Everything is moved to left side.
Screenshot_157
Screenshot_158

**Comment by rich20** [comment]
Many more files have been changed. I have replied to you in the forum. https://www.kunena.org/forum/36-German/166363-anmeldung-und-darstellungsprobleme-aurelia-vorlage-kunena-6-0-2#226026

**Pull request closed without merging**